src/main/java/org/brapi/test/BrAPITestServer/factory/core/TrialComponent.java

```diff
@@ -0,0 +1,54 @@
+package org.brapi.test.BrAPITestServer.factory.core;
+
+import io.swagger.model.Metadata;
+import io.swagger.model.core.BatchTypes;
+import io.swagger.model.core.Trial;
+import io.swagger.model.core.TrialSearchRequest;
+import org.brapi.test.BrAPITestServer.factory.BrAPIComponent;
+import org.brapi.test.BrAPITestServer.model.entity.SearchRequestEntity;
+import org.brapi.test.BrAPITestServer.service.core.TrialService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class TrialComponent implements BrAPIComponent<Trial, TrialSearchRequest> {
+    private final TrialService trialService;
+
+    @Autowired
+    public TrialComponent(TrialService trialService) {
+        this.trialService = trialService;
+    }
+
+    @Override
+    public List<Trial> findEntities(TrialSearchRequest request, Metadata metadata) {
+        return trialService.findTrials(request, metadata);
+    }
+
+    @Override
+    public BatchTypes getBatchType() {
+        return BatchTypes.TRIALS;
+    }
+
+    @Override
+    public SearchRequestEntity.SearchRequestTypes getSearchType() {
```

Comment:
This method is never used. please remove (see comment in BrAPIComponent.java

Reply:
done

Reply:
Marking resolved - this was removed